### PR TITLE
Starting with pymmcore-plus implementation of iSIM control

### PR DESCRIPTION
**Pull request by wl-stepp**
I have started with getting the iSIM to work on pymmcore-plus.
Goal would be to first be able to run an arbitrary MDASequence provided by useq. This would allow to use the maximum amount of functionality from pymmcore-plus. This would be achieved by a custom MDAEngine that manages both MDASequence and MDAEvent information to execute an incoming MDAEvent.
Additionally there is a LiveEngine, that runs a live mode for scanning the sample in iSIM mode.

Open questions:
* Most information necessary for the corresponding ni_data can be taken from the MDAEvent directly, except for the laser powers to be sent to the AOTF. These will have to be passed some other way. How?
* It's many times requested to have different settings for "live" and "acquisitions". How would we handle that in logic and gui?

### CODE REVIEW

File: control/ni/testing/triggered_camera.py
Comment (wl-stepp) [inline control/ni/testing/triggered_camera.py:1]: I get down to 145 ms cycle time for an exposure time set in the camera of 100ms. This works out nicely with the delays that we need in the iSIM scanning cycle.

File: control/ni/devices.py [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+    #     n_shift = (round(self.settings.camera_readout_time * self.settings.final_sample_rate) -
+    #                round(self.settings.final_sample_rate * self.adjusted_readout))
+    #     print("LED readout")
+    #     print(n_shift)
+    #     readout_delay0 = np.zeros((frame.shape[0],
+    #                                round(self.settings.final_sample_rate * self.settings.camera_readout_time) - n_shift))
+    #     readout_delay1 = np.zeros((frame.shape[0],
+    #                                round(self.settings.final_sample_rate * self.settings.camera_readout_time)))
+    #     frame = np.hstack([readout_delay0, frame, readout_delay1])
+    #     return frame
```
Comment (kmdouglass) [inline control/ni/devices.py:147]: Can we either delete these comments or uncomment them?

**Review by kmdouglass — COMMENTED** [review]
>     * Most information necessary for the corresponding ni_data can be taken from the MDAEvent directly, except for the laser powers to be sent to the AOTF. These will have to be passed some other way. How?

One way to think about this is to think about the data that "flows" through the program. If you separate these two data sources (laser powers and MDAEvents), then you create more channels than I think is necessary. Instead, you could create your own dictionary-like object (maybe a `TypedDict`) that contains all the necessary info for the engine to do something. This could wrap a `MDAEvent`, or be converted into one to hand off to `pymmcore-plus`, and also be converted into another set of data to hand off to the NIDAQ.

An advantage of this is that you would define the core data type, rather than pymmcore-plus, and this would help create a degree of decoupling from it.

**Comment by kmdouglass** [comment]
Feel free to open the PR now so we can merge and start testing it. We don't need everything to be completely working yet. It will also help keep PR sizes smaller in the future.

File: control/ni/devices.py (hunk 1 above)
Comment (wl-stepp) [reply control/ni/devices.py:147]: The LED is not yet adapted from the original version. Will do that tomorrow and uncomment.

**Review by kmdouglass — APPROVED** [review]
Ship it. :shipit: ❤️ 

We can continue to sort out the architecture in follow up PRs.

**Pull request merged**